test/registry-setup/main.go

```diff
@@ -0,0 +1,283 @@
+
+func main() {
+	flag.Parse()
+
+	if *configmap == "" {
+		fmt.Fprintf(os.Stderr, "ConfigMap file name can not be empty")
+		os.Exit(1)
+	}
+
+	work := func(loader Loader) error {
+		cm, err := read(*configmap)
+		if err != nil {
+			return err
+		}
+
+		if err = loader(cm); err != nil {
+			return err
+		}
+
+		return write(*configmap, cm)
+	}
+
+	l := func(mode string, isOLM bool) (loader Loader, err error) {
+		switch {
+		case mode == "ci" && isOLM:
+			loader = withOLMWithCI
+		case mode == "ci":
+			loader = withCI
+		case mode == "local" && isOLM:
+			loader = withOLMWithLocal
+		case mode == "local":
+			loader = withLocal
+		default:
+			err = fmt.Errorf("unsupported mode, value=%s", mode)
+		}
+
+		return
+	}
+
+	loader, err := l(*mode, *isOLM)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ran in to error - %s", err.Error())
+		os.Exit(1)
+	}
+
+	if err = work(loader); err != nil {
+		fmt.Fprintf(os.Stderr, "ran in to error - %s", err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Fprintf(os.Stderr, "successfully updated ConfigMap env file=%s", *configmap)
+	os.Exit(0)
+}
+
+func withOLMWithLocal(object *corev1.ConfigMap) error {
+	if err := populate(getLocalImageURL, object); err != nil {
+		return err
+	}
+
+	return validateWithOLM(object)
+}
+
+func withOLMWithCI(object *corev1.ConfigMap) error {
+	if err := populate(getCIImageURL, object); err != nil {
+		return err
+	}
+
+	return validateWithOLM(object)
+}
+
+func withLocal(object *corev1.ConfigMap) error {
+	if err := populate(getLocalImageURL, object); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func withCI(object *corev1.ConfigMap) error {
+	if err := populate(getCIImageURL, object); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateWithOLM(object *corev1.ConfigMap) error {
+	if object.Data[oldOperatorImageEnv] == "" {
+		return fmt.Errorf("ENV var %s not defined", oldOperatorImageEnv)
+	}
+
+	if object.Data[oldOperandImageEnv] == "" {
+		return fmt.Errorf("ENV var %s not defined", oldOperandImageEnv)
+	}
+
+	if object.Data[csvFilePathEnv] == "" {
+		return fmt.Errorf("ENV var %s not defined", csvFilePathEnv)
+	}
+
+	if object.Data[operatorImageEnv] == "" {
+		return fmt.Errorf("ENV var %s not defined", operatorImageEnv)
+	}
+
+	if object.Data[operandImageEnv] == "" {
+		return fmt.Errorf("ENV var %s not defined", operandImageEnv)
+	}
+
+	if object.Data[operatorRegistryImageEnv] == "" {
+		return fmt.Errorf("ENV var %s not defined", operatorRegistryImageEnv)
+	}
+
+	return nil
+}
+
+// This function populates the ConfigMap object's data section with all the ENV variables
+// so that we can propagate these ENV variables to the init container.
+func populate(getter ImageURLGetter, object *corev1.ConfigMap) error {
+	image, err := getter()
+	if err != nil {
+		return err
+	}
+
+	oldOperatorImageURL := os.Getenv(oldOperatorImageEnv)
+	oldOperandImageURL := os.Getenv(oldOperandImageEnv)
+	csvFilePath := os.Getenv(csvFilePathEnv)
+
+	if len(object.Data) == 0 {
```

Comment:
this implies the object already exists, I don't believe we need the object.Data = map[string]string{} on the next line